Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM golang:1.9 AS builder
+
+WORKDIR /go/src/github.com/syncthing/syncthing
+COPY . .
+
+ENV CGO_ENABLED=0
+ENV BUILD_HOST=syncthing.net
```

Comment:
BUILD_HOST and BUILD_USER don't work in my environment, the resulting syncthing binary announces itself as built by unknown-user:
` sudo docker run -it --rm --entrypoint syncthing newsyncthing -version
syncthing v0.14.44-rc.2+24-g1aabe427-docker "Dysprosium Dragonfly" (go1.9.3 linux-amd64) unknown-user@8459f1a59c00 2018-02-04 14:27:14 UTC`

Reply:
Only way I can imagine this failing is if the binary was already in place and up to date so the build.go script doesn't rebuild it... adding an rm to the Dockerfile to counter that scenario

Reply:
OK, facepalm, I'm not awake. You can discard my comment. I was on the wrong docker branch.

Reply:
Oh right, good, I pushed an early version to the main repo to test an automated build, yes. Will remove to avoid confusion :D